controllers/snow_machineconfig_controller.go

```diff
@@ -75,36 +74,34 @@ func (r *SnowMachineConfigReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 	result, err := r.reconcile(ctx, snowMachineConfig)
 	if err != nil {
-		log.Error(err, "Failed to reconcile SnowMachineConfig")
-		reterr = kerrors.NewAggregate([]error{reterr, err})
+		reterr = kerrors.NewAggregate([]error{reterr, fmt.Errorf("Failed to reconcile snowmachineconfig: %v", err)})
 	}
 	return result, reterr
 }
 
 func (r *SnowMachineConfigReconciler) reconcile(ctx context.Context, snowMachineConfig *anywherev1.SnowMachineConfig) (_ ctrl.Result, reterr error) {
 	// TODO: need to figure out how to load creds in controller
+	var allErrs []error
 	deviceClientMap, err := r.clientBuilder.BuildSnowAwsClientMap(ctx)
 	if err != nil {
-		failureMessage := err.Error()
-		snowMachineConfig.Status.FailureMessage = &failureMessage
 		return ctrl.Result{}, err
 	}
 	// Setting the aws client map on every reconcile based on the secrets at that point of time
 	validator := snow.NewValidator(deviceClientMap)
 	if err := validator.ValidateMachineDeviceIPs(ctx, snowMachineConfig); err != nil {
-		failureMessage := err.Error()
-		snowMachineConfig.Status.FailureMessage = &failureMessage
-		return ctrl.Result{}, err
+		allErrs = append(allErrs, err)
 	}
 	if err := validator.ValidateEC2ImageExistsOnDevice(ctx, snowMachineConfig); err != nil {
-		failureMessage := err.Error()
-		snowMachineConfig.Status.FailureMessage = &failureMessage
-		return ctrl.Result{}, err
+		allErrs = append(allErrs, err)
 	}
 	if err := validator.ValidateEC2SshKeyNameExists(ctx, snowMachineConfig); err != nil {
-		failureMessage := err.Error()
+		allErrs = append(allErrs, err)
+	}
+	if len(allErrs) > 0 {
+		aggregate := kerrors.NewAggregate(allErrs)
+		failureMessage := aggregate.Error()
```

Comment:
I would set `snowMachineConfig.Status.SpecValid = false`
It's true that's the default, but it can go from valid to invalid